src/graphics/draw.rs

```diff
@@ -254,6 +254,18 @@ where
     }
 }
 
+/// All types that can be drawn onto a canvas implement the `Drawable` trait.
+pub trait Drawable {
+    /// Draws the drawable onto the canvas.
+    fn draw(self, canvas: &mut Canvas, param: DrawParam);
+
+    /// Returns a bounding box in the form of a `Rect`.
+    ///
+    /// It returns `Option` because some `Drawable`s may have no bounding box,
+    /// namely `InstanceArray` (as there is no true bounds for the instances given the instanced mesh can differ).
```

Comment:
Ah, ok, now I see why it returns `None`.
Hmm... we could remove `dimensions` from this trait and then just give `InstanceArray` it's own `dimension` function that also takes an `Option<Mesh>`. Or maybe that's not worth it...